core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -415,4 +458,29 @@ public void close(ScheduledExecutorService instance) {
           instance.shutdown();
         }
       };
+
+  private static final class InactiveTransport implements ClientTransport {
+    private static final InactiveTransport INSTANCE = new InactiveTransport();
+
+    @Override
+    public ClientStream newStream(
+        MethodDescriptor<?, ?> method, Headers headers, ClientStreamListener listener) {
+      throw new IllegalStateException("Transport is inactive");
+    }
+
+    @Override
+    public void start(Listener listener) {
+      throw new IllegalStateException();
+    }
+
+    @Override
+    public void ping(PingCallback callback, Executor executor) {
+      throw new IllegalStateException("Transport is inactive");
```

Comment:
This should not throw, and instead transient failures are supposed to be reported to the `callback`. Using the status passed to `transportShutdown()` seems very appropriate.

```java
  executor.execute(new Runnable() {
    @Override
    public void run() {
      callback.pingFailed(status.asException()); // or asRuntimeException()
    }
  });
```